### DISCUSSION TITLE
Vits-TTS-finetune on pretrained model

### DISCUSSION THREAD

**Discussion by madey83** [idiap/coqui-ai-TTS#151] [Q&A] [answered]
When i executed this only  python.exe .\_train_vits_test_windows.py training begins...

When i have follow this https://docs.coqui.ai/en/latest/finetuning.html i heave got this problem:

python.exe .\_train_vits_test_windows.py --config_path D:\TTS-Training\tts_models--pl--mai_female--vits\model_file.pth.tar --coqpit.run_name "Vits-TTS-PL-finetune" --coqpit.lr 0.00001
usage: _train_vits_test_windows.py [-h] [--coqpit.output_path COQPIT.OUTPUT_PATH] [--coqpit.run_name COQPIT.RUN_NAME]
                                   [--coqpit.run_description COQPIT.RUN_DESCRIPTION]
                                   [--coqpit.print_step COQPIT.PRINT_STEP] [--coqpit.plot_step COQPIT.PLOT_STEP]
                                   [--coqpit.model_param_stats true/false]
                                   [--coqpit.dashboard_logger COQPIT.DASHBOARD_LOGGER]
                                   [--coqpit.save_on_interrupt true/false] [--coqpit.save_step COQPIT.SAVE_STEP]
                                   [--coqpit.save_n_checkpoints COQPIT.SAVE_N_CHECKPOINTS]
                                   [--coqpit.save_checkpoints true/false] [--coqpit.save_all_best true/false]
                                   [--coqpit.save_best_after COQPIT.SAVE_BEST_AFTER] [--coqpit.print_eval true/false]
                                   [--coqpit.test_delay_epochs COQPIT.TEST_DELAY_EPOCHS]
                                   [--coqpit.run_eval true/false]
                                   [--coqpit.distributed_backend COQPIT.DISTRIBUTED_BACKEND]
                                   [--coqpit.distributed_url COQPIT.DISTRIBUTED_URL]
                                   [--coqpit.mixed_precision true/false] [--coqpit.precision COQPIT.PRECISION]
                                   [--coqpit.epochs COQPIT.EPOCHS] [--coqpit.batch_size COQPIT.BATCH_SIZE]
                                   [--coqpit.eval_batch_size COQPIT.EVAL_BATCH_SIZE]
                                   [--coqpit.grad_clip.0 COQPIT.GRAD_CLIP.0] [--coqpit.grad_clip.1 COQPIT.GRAD_CLIP.1]
                                   [--coqpit.scheduler_after_epoch true/false] [--coqpit.optimizer COQPIT.OPTIMIZER]
                                   [--coqpit.optimizer_params COQPIT.OPTIMIZER_PARAMS]
                                   [--coqpit.lr_scheduler COQPIT.LR_SCHEDULER]
                                   [--coqpit.lr_scheduler_params COQPIT.LR_SCHEDULER_PARAMS]
                                   [--coqpit.use_grad_scaler true/false] [--coqpit.allow_tf32 true/false]
                                   [--coqpit.cudnn_enable true/false] [--coqpit.cudnn_deterministic true/false]
                                   [--coqpit.cudnn_benchmark true/false] [--coqpit.training_seed COQPIT.TRAINING_SEED]
                                   [--coqpit.model COQPIT.MODEL]
                                   [--coqpit.num_loader_workers COQPIT.NUM_LOADER_WORKERS]
                                   [--coqpit.num_eval_loader_workers COQPIT.NUM_EVAL_LOADER_WORKERS]
                                   [--coqpit.use_noise_augment true/false]
                                   [--coqpit.audio.fft_size COQPIT.AUDIO.FFT_SIZE]
                                   [--coqpit.audio.sample_rate COQPIT.AUDIO.SAMPLE_RATE]
                                   [--coqpit.audio.win_length COQPIT.AUDIO.WIN_LENGTH]
                                   [--coqpit.audio.hop_length COQPIT.AUDIO.HOP_LENGTH]
                                   [--coqpit.audio.num_mels COQPIT.AUDIO.NUM_MELS]
                                   [--coqpit.audio.mel_fmin COQPIT.AUDIO.MEL_FMIN]
                                   [--coqpit.audio.mel_fmax COQPIT.AUDIO.MEL_FMAX] [--coqpit.use_phonemes true/false]
                                   [--coqpit.phonemizer COQPIT.PHONEMIZER]
                                   [--coqpit.phoneme_language COQPIT.PHONEME_LANGUAGE]
                                   [--coqpit.compute_input_seq_cache true/false]
                                   [--coqpit.text_cleaner COQPIT.TEXT_CLEANER]
                                   [--coqpit.enable_eos_bos_chars true/false]
                                   [--coqpit.test_sentences_file COQPIT.TEST_SENTENCES_FILE]
                                   [--coqpit.phoneme_cache_path COQPIT.PHONEME_CACHE_PATH]
                                   [--coqpit.characters.characters_class COQPIT.CHARACTERS.CHARACTERS_CLASS]
                                   [--coqpit.characters.pad COQPIT.CHARACTERS.PAD]
                                   [--coqpit.characters.eos COQPIT.CHARACTERS.EOS]
                                   [--coqpit.characters.bos COQPIT.CHARACTERS.BOS]
                                   [--coqpit.characters.blank COQPIT.CHARACTERS.BLANK]
                                   [--coqpit.characters.characters COQPIT.CHARACTERS.CHARACTERS]
                                   [--coqpit.characters.punctuations COQPIT.CHARACTERS.PUNCTUATIONS]
                                   [--coqpit.characters.phonemes COQPIT.CHARACTERS.PHONEMES]
                                   [--coqpit.characters.is_unique true/false]
                                   [--coqpit.characters.is_sorted true/false] [--coqpit.add_blank true/false]
                                   [--coqpit.batch_group_size COQPIT.BATCH_GROUP_SIZE]
                                   [--coqpit.loss_masking true/false] [--coqpit.min_audio_len COQPIT.MIN_AUDIO_LEN]
                                   [--coqpit.max_audio_len COQPIT.MAX_AUDIO_LEN]
                                   [--coqpit.min_text_len COQPIT.MIN_TEXT_LEN]
                                   [--coqpit.max_text_len COQPIT.MAX_TEXT_LEN] [--coqpit.compute_f0 true/false]
                                   [--coqpit.compute_energy true/false] [--coqpit.compute_linear_spec true/false]
                                   [--coqpit.precompute_num_workers COQPIT.PRECOMPUTE_NUM_WORKERS]
                                   [--coqpit.start_by_longest true/false] [--coqpit.shuffle true/false]
                                   [--coqpit.drop_last true/false]
                                   [--coqpit.datasets.0.formatter COQPIT.DATASETS.0.FORMATTER]
                                   [--coqpit.datasets.0.dataset_name COQPIT.DATASETS.0.DATASET_NAME]
                                   [--coqpit.datasets.0.path COQPIT.DATASETS.0.PATH]
                                   [--coqpit.datasets.0.meta_file_train COQPIT.DATASETS.0.META_FILE_TRAIN]
                                   [--coqpit.datasets.0.ignored_speakers [COQPIT.DATASETS.0.IGNORED_SPEAKERS ...]]
                                   [--coqpit.datasets.0.language COQPIT.DATASETS.0.LANGUAGE]
                                   [--coqpit.datasets.0.phonemizer COQPIT.DATASETS.0.PHONEMIZER]
                                   [--coqpit.datasets.0.meta_file_val COQPIT.DATASETS.0.META_FILE_VAL]
                                   [--coqpit.datasets.0.meta_file_attn_mask COQPIT.DATASETS.0.META_FILE_ATTN_MASK]
                                   [--coqpit.eval_split_max_size COQPIT.EVAL_SPLIT_MAX_SIZE]
                                   [--coqpit.eval_split_size COQPIT.EVAL_SPLIT_SIZE]
                                   [--coqpit.use_speaker_weighted_sampler true/false]
                                   [--coqpit.speaker_weighted_sampler_alpha COQPIT.SPEAKER_WEIGHTED_SAMPLER_ALPHA]
                                   [--coqpit.use_language_weighted_sampler true/false]
                                   [--coqpit.language_weighted_sampler_alpha COQPIT.LANGUAGE_WEIGHTED_SAMPLER_ALPHA]
                                   [--coqpit.use_length_weighted_sampler true/false]
                                   [--coqpit.length_weighted_sampler_alpha COQPIT.LENGTH_WEIGHTED_SAMPLER_ALPHA]
                                   [--coqpit.model_args.num_chars COQPIT.MODEL_ARGS.NUM_CHARS]
                                   [--coqpit.model_args.out_channels COQPIT.MODEL_ARGS.OUT_CHANNELS]
                                   [--coqpit.model_args.spec_segment_size COQPIT.MODEL_ARGS.SPEC_SEGMENT_SIZE]
                                   [--coqpit.model_args.hidden_channels COQPIT.MODEL_ARGS.HIDDEN_CHANNELS]
                                   [--coqpit.model_args.hidden_channels_ffn_text_encoder COQPIT.MODEL_ARGS.HIDDEN_CHANNELS_FFN_TEXT_ENCODER]
                                   [--coqpit.model_args.num_heads_text_encoder COQPIT.MODEL_ARGS.NUM_HEADS_TEXT_ENCODER]
                                   [--coqpit.model_args.num_layers_text_encoder COQPIT.MODEL_ARGS.NUM_LAYERS_TEXT_ENCODER]
                                   [--coqpit.model_args.kernel_size_text_encoder COQPIT.MODEL_ARGS.KERNEL_SIZE_TEXT_ENCODER]
                                   [--coqpit.model_args.dropout_p_text_encoder COQPIT.MODEL_ARGS.DROPOUT_P_TEXT_ENCODER]
                                   [--coqpit.model_args.dropout_p_duration_predictor COQPIT.MODEL_ARGS.DROPOUT_P_DURATION_PREDICTOR]
                                   [--coqpit.model_args.kernel_size_posterior_encoder COQPIT.MODEL_ARGS.KERNEL_SIZE_POSTERIOR_ENCODER]
                                   [--coqpit.model_args.dilation_rate_posterior_encoder COQPIT.MODEL_ARGS.DILATION_RATE_POSTERIOR_ENCODER]
                                   [--coqpit.model_args.num_layers_posterior_encoder COQPIT.MODEL_ARGS.NUM_LAYERS_POSTERIOR_ENCODER]
                                   [--coqpit.model_args.kernel_size_flow COQPIT.MODEL_ARGS.KERNEL_SIZE_FLOW]
                                   [--coqpit.model_args.dilation_rate_flow COQPIT.MODEL_ARGS.DILATION_RATE_FLOW]
                                   [--coqpit.model_args.num_layers_flow COQPIT.MODEL_ARGS.NUM_LAYERS_FLOW]
                                   [--coqpit.model_args.resblock_type_decoder COQPIT.MODEL_ARGS.RESBLOCK_TYPE_DECODER]
                                   [--coqpit.model_args.resblock_kernel_sizes_decoder.0 COQPIT.MODEL_ARGS.RESBLOCK_KERNEL_SIZES_DECODER.0]
                                   [--coqpit.model_args.resblock_kernel_sizes_decoder.1 COQPIT.MODEL_ARGS.RESBLOCK_KERNEL_SIZES_DECODER.1]
                                   [--coqpit.model_args.resblock_kernel_sizes_decoder.2 COQPIT.MODEL_ARGS.RESBLOCK_KERNEL_SIZES_DECODER.2]
                                   [--coqpit.model_args.upsample_rates_decoder.0 COQPIT.MODEL_ARGS.UPSAMPLE_RATES_DECODER.0]
                                   [--coqpit.model_args.upsample_rates_decoder.1 COQPIT.MODEL_ARGS.UPSAMPLE_RATES_DECODER.1]
                                   [--coqpit.model_args.upsample_rates_decoder.2 COQPIT.MODEL_ARGS.UPSAMPLE_RATES_DECODER.2]
                                   [--coqpit.model_args.upsample_rates_decoder.3 COQPIT.MODEL_ARGS.UPSAMPLE_RATES_DECODER.3]
                                   [--coqpit.model_args.upsample_initial_channel_decoder COQPIT.MODEL_ARGS.UPSAMPLE_INITIAL_CHANNEL_DECODER]
                                   [--coqpit.model_args.upsample_kernel_sizes_decoder.0 COQPIT.MODEL_ARGS.UPSAMPLE_KERNEL_SIZES_DECODER.0]
                                   [--coqpit.model_args.upsample_kernel_sizes_decoder.1 COQPIT.MODEL_ARGS.UPSAMPLE_KERNEL_SIZES_DECODER.1]
                                   [--coqpit.model_args.upsample_kernel_sizes_decoder.2 COQPIT.MODEL_ARGS.UPSAMPLE_KERNEL_SIZES_DECODER.2]
                                   [--coqpit.model_args.upsample_kernel_sizes_decoder.3 COQPIT.MODEL_ARGS.UPSAMPLE_KERNEL_SIZES_DECODER.3]
                                   [--coqpit.model_args.periods_multi_period_discriminator.0 COQPIT.MODEL_ARGS.PERIODS_MULTI_PERIOD_DISCRIMINATOR.0]
                                   [--coqpit.model_args.periods_multi_period_discriminator.1 COQPIT.MODEL_ARGS.PERIODS_MULTI_PERIOD_DISCRIMINATOR.1]
                                   [--coqpit.model_args.periods_multi_period_discriminator.2 COQPIT.MODEL_ARGS.PERIODS_MULTI_PERIOD_DISCRIMINATOR.2]
                                   [--coqpit.model_args.periods_multi_period_discriminator.3 COQPIT.MODEL_ARGS.PERIODS_MULTI_PERIOD_DISCRIMINATOR.3]
                                   [--coqpit.model_args.periods_multi_period_discriminator.4 COQPIT.MODEL_ARGS.PERIODS_MULTI_PERIOD_DISCRIMINATOR.4]
                                   [--coqpit.model_args.use_sdp true/false]
                                   [--coqpit.model_args.noise_scale COQPIT.MODEL_ARGS.NOISE_SCALE]
                                   [--coqpit.model_args.inference_noise_scale COQPIT.MODEL_ARGS.INFERENCE_NOISE_SCALE]
                                   [--coqpit.model_args.length_scale COQPIT.MODEL_ARGS.LENGTH_SCALE]
                                   [--coqpit.model_args.noise_scale_dp COQPIT.MODEL_ARGS.NOISE_SCALE_DP]
                                   [--coqpit.model_args.inference_noise_scale_dp COQPIT.MODEL_ARGS.INFERENCE_NOISE_SCALE_DP]
                                   [--coqpit.model_args.max_inference_len COQPIT.MODEL_ARGS.MAX_INFERENCE_LEN]
                                   [--coqpit.model_args.init_discriminator true/false]
                                   [--coqpit.model_args.use_spectral_norm_disriminator true/false]
                                   [--coqpit.model_args.use_speaker_embedding true/false]
                                   [--coqpit.model_args.num_speakers COQPIT.MODEL_ARGS.NUM_SPEAKERS]
                                   [--coqpit.model_args.speakers_file COQPIT.MODEL_ARGS.SPEAKERS_FILE]
                                   [--coqpit.model_args.d_vector_file [COQPIT.MODEL_ARGS.D_VECTOR_FILE ...]]
                                   [--coqpit.model_args.speaker_embedding_channels COQPIT.MODEL_ARGS.SPEAKER_EMBEDDING_CHANNELS]
                                   [--coqpit.model_args.use_d_vector_file true/false]
                                   [--coqpit.model_args.d_vector_dim COQPIT.MODEL_ARGS.D_VECTOR_DIM]
                                   [--coqpit.model_args.detach_dp_input true/false]
                                   [--coqpit.model_args.use_language_embedding true/false]
                                   [--coqpit.model_args.embedded_language_dim COQPIT.MODEL_ARGS.EMBEDDED_LANGUAGE_DIM]
                                   [--coqpit.model_args.num_languages COQPIT.MODEL_ARGS.NUM_LANGUAGES]
                                   [--coqpit.model_args.language_ids_file COQPIT.MODEL_ARGS.LANGUAGE_IDS_FILE]
                                   [--coqpit.model_args.use_speaker_encoder_as_loss true/false]
                                   [--coqpit.model_args.speaker_encoder_config_path COQPIT.MODEL_ARGS.SPEAKER_ENCODER_CONFIG_PATH]
                                   [--coqpit.model_args.speaker_encoder_model_path COQPIT.MODEL_ARGS.SPEAKER_ENCODER_MODEL_PATH]
                                   [--coqpit.model_args.condition_dp_on_speaker true/false]
                                   [--coqpit.model_args.freeze_encoder true/false]
                                   [--coqpit.model_args.freeze_DP true/false]
                                   [--coqpit.model_args.freeze_PE true/false]
                                   [--coqpit.model_args.freeze_flow_decoder true/false]
                                   [--coqpit.model_args.freeze_waveform_decoder true/false]
                                   [--coqpit.model_args.encoder_sample_rate COQPIT.MODEL_ARGS.ENCODER_SAMPLE_RATE]
                                   [--coqpit.model_args.interpolate_z true/false]
                                   [--coqpit.model_args.reinit_DP true/false]
                                   [--coqpit.model_args.reinit_text_encoder true/false]
                                   [--coqpit.lr_gen COQPIT.LR_GEN] [--coqpit.lr_disc COQPIT.LR_DISC]
                                   [--coqpit.lr_scheduler_gen COQPIT.LR_SCHEDULER_GEN]
                                   [--coqpit.lr_scheduler_gen_params COQPIT.LR_SCHEDULER_GEN_PARAMS]
                                   [--coqpit.lr_scheduler_disc COQPIT.LR_SCHEDULER_DISC]
                                   [--coqpit.lr_scheduler_disc_params COQPIT.LR_SCHEDULER_DISC_PARAMS]
                                   [--coqpit.kl_loss_alpha COQPIT.KL_LOSS_ALPHA]
                                   [--coqpit.disc_loss_alpha COQPIT.DISC_LOSS_ALPHA]
                                   [--coqpit.gen_loss_alpha COQPIT.GEN_LOSS_ALPHA]
                                   [--coqpit.feat_loss_alpha COQPIT.FEAT_LOSS_ALPHA]
                                   [--coqpit.mel_loss_alpha COQPIT.MEL_LOSS_ALPHA]
                                   [--coqpit.dur_loss_alpha COQPIT.DUR_LOSS_ALPHA]
                                   [--coqpit.speaker_encoder_loss_alpha COQPIT.SPEAKER_ENCODER_LOSS_ALPHA]
                                   [--coqpit.return_wav true/false] [--coqpit.use_weighted_sampler true/false]
                                   [--coqpit.weighted_sampler_attrs COQPIT.WEIGHTED_SAMPLER_ATTRS]
                                   [--coqpit.weighted_sampler_multipliers COQPIT.WEIGHTED_SAMPLER_MULTIPLIERS]
                                   [--coqpit.r COQPIT.R] [--coqpit.num_speakers COQPIT.NUM_SPEAKERS]
                                   [--coqpit.use_speaker_embedding true/false]
                                   [--coqpit.speakers_file COQPIT.SPEAKERS_FILE]
                                   [--coqpit.speaker_embedding_channels COQPIT.SPEAKER_EMBEDDING_CHANNELS]
                                   [--coqpit.language_ids_file COQPIT.LANGUAGE_IDS_FILE]
                                   [--coqpit.use_language_embedding true/false]
                                   [--coqpit.use_d_vector_file true/false]
                                   [--coqpit.d_vector_file [COQPIT.D_VECTOR_FILE ...]]
                                   [--coqpit.d_vector_dim COQPIT.D_VECTOR_DIM]
_train_vits_test_windows.py: error: ambiguous option: --coqpit.lr could match --coqpit.lr_scheduler, --coqpit.lr_scheduler_params, --coqpit.lr_gen, --coqpit.lr_disc, --coqpit.lr_scheduler_gen, --coqpit.lr_scheduler_gen_params, --coqpit.lr_scheduler_disc, --coqpit.lr_scheduler_disc_params

Can i ask for help?

**Comment by eginhard** [comment, marked as answer]
As the error message shows, Vits models don't have a `--coqpit.lr` option. You need to set `--copit.lr_gen` and/or `--coqpit.lr_disc` or just leave them out.

**Reply by madey83** [reply]
thank you

**Comment by madey83** [comment]
done
